src/flb_pack_gelf.c

```diff
@@ -548,17 +549,38 @@ flb_sds_t flb_msgpack_to_gelf(flb_sds_t *s, msgpack_object *o,
                 if (v->type == MSGPACK_OBJECT_POSITIVE_INTEGER) {
                     if ( v->via.u64 > 7 ) {
                         flb_error("[flb_msgpack_to_gelf] level is %" PRIu64 ", "
-                                  "but should be in 0..7", v->via.u64);
+                                  "but should be in 0..7 or a syslog keyword", v->via.u64);
                         return NULL;
                     }
                 }
                 else if (v->type == MSGPACK_OBJECT_STR) {
                     val     = v->via.str.ptr;
                     val_len = v->via.str.size;
-                    if (val_len != 1 || val[0] < '0' || val[0] > '7') {
-                        flb_error("[flb_msgpack_to_gelf] level is '%.*s', "
-                                  "but should be in 0..7", val_len, val);
-                        return NULL;
+                    if (val_len == 1 && val[0] >= '0' && val[0] <= '7') {
+                        v = &vtmp;
+                        v->type = MSGPACK_OBJECT_POSITIVE_INTEGER;
+                        v->via.u64 = (uint64_t)(val[0] - '0');
+                    }
+                    else {
+                        int i;
```

Comment:
It's confusing, at least for me, use the same variable of the upper loop

Reply:
Ah, I didn't mention the upper level `i`. Now it's confusing for me too!